google-cloud-nio/src/main/resources/META-INF/native-image/com/google/cloud/google-cloud-nio/native-image.properties

```
@@ -0,0 +1,38 @@
+# Using META-INF/services with Native Image compilation results in
+# FileSystemProvider being initialized at build time. This results
+# CloudStorageFileSystemProvider and some classes referenced by
+# this class (for example, StorageOptions, StorageOptionsUtil,
+# DefaultStorageRetryStrategy) being unexpectedly and recursively initialized at
+# build time.
+Args = --initialize-at-build-time=com.google.cloud.storage.contrib.nio.CloudStorageFileSystemProvider,\
+  com.google.cloud.storage.contrib.nio.CloudStorageConfiguration,\
+  com.google.cloud.storage.contrib.nio.CloudStorageFileSystem,\
+  com.google.cloud.storage.contrib.nio.StorageOptionsUtil,\
+  com.google.cloud.storage.StorageException,\
+  com.google.cloud.storage.StorageOptions,\
+  com.google.cloud.storage.DefaultStorageRetryStrategy,\
+  com.google.api.client.http.OpenCensusUtils,\
+  com.google.api.client.http.GenericUrl,\
+  com.google.api.client.http.HttpRequest,\
+  com.google.api.client.http.HttpMediaType,\
+  com.google.api.client.http.javanet.NetHttpTransport,\
+  com.google.api.client.util.Data,\
+  com.google.api.client.util.DateTime,\
+  com.google.auth.oauth2,\
+  io.opencensus.trace,\
+  com.google.api.client.util.escape.CharEscapers,\
+  com.google.api.client.util.escape.PercentEscaper,\
+  io.opencensus.contrib.http.util.CloudTraceFormat,\
+  io.grpc.Context,\
+  io.grpc.Context$LazyStorage,\
+  io.grpc.ThreadLocalContextStorage,\
+  com.google.common.cache.LocalCache,\
+  com.google.common.cache.CacheBuilder,\
+  com.google.common.math.IntMath$1,\
+  com.google.common.collect.RegularImmutableMap,\
+  com.google.gson.stream.JsonReader,\
+  com.google.api.client.json.gson.GsonParser$1,\
+  com.google.api.client.json.JsonParser$1,\
+  com.google.cloud.PlatformInformation,\
+  com.google.cloud.ExceptionHandler,\
+  org.threeten.bp.Duration
```

I had to add the following additional classes when running via
```bash
export JAVA_HOME=~/opt/java/11-native
mvn -Pnative clean verify
```

```bash
> $JAVA_HOME/bin/java -version
openjdk version "11.0.14" 2022-01-18
OpenJDK Runtime Environment GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05)
OpenJDK 64-Bit Server VM GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05, mixed mode, sharing)
```

```suggestion
  org.threeten.bp.Duration,\
  com.google.common.io.BaseEncoding$Base16Encoding,\
  com.google.common.io.BaseEncoding$Base64Encoding,\
  com.google.common.io.BaseEncoding,\
  com.google.common.io.BaseEncoding$StandardBaseEncoding,\
  com.google.api.client.util.PemReader

```

---

Thank you for verifying! Added these classes in. I had to skip running few tests due to project setup related issues so may have missed these when I commented them out.

---

Ah okay, thanks for confirming